test/test_main.py

```diff
@@ -99,25 +115,37 @@ def _read_and_clear_state(self):
                 state_file.truncate()
         return state
 
-    def _simulate_event(self, event_name, charm_config):
-        event_hook = JUJU_CHARM_DIR / f"hooks/{event_name.replace('_', '-')}"
+    def _simulate_event(self, event_spec, charm_config):
```

Comment:
Looks like charm_config should be part of the event_spec.

Reply:
This introduces a requirement of passing it into every `EventSpec` constructor but it does seem to belong there.